en/guide/extending_csd.md

```diff
@@ -1,6 +1,121 @@
 # Extending Camera Streaming Daemon
 
```

Comment:
Worth creating a (short) overview of what can be extended - essentially support for new cameras and support for new streams. 

Would this be a good place for your diagram (https://github.com/lbegani/camera-streaming-daemon/blob/07ed852eccea5222eb0e721a8659f4d4a7db527c/docs/assets/camera_server_class_diagram.png) ... possibly changing the appearance of the classes that need to be derived from to make them more obvious - maybe adding a dummy derived class for that too to the diagram. ?

Reply:
1. Shall add a list of extensible items
2. Add a class diagram showing base `CameraDevice` Class and derived `CameraDeviceCustom` class